components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -67,6 +66,7 @@ void RewardsNotificationServiceImpl::AddNotification(
   RewardsNotification rewards_notification(
       id, type, GenerateRewardsNotificationTimestamp(), std::move(args));
   rewards_notifications_[id] = rewards_notification;
+  StoreRewardsNotifications();
```

Comment:
why this change?

Reply:
I guess we never store a notification when we add it. On desktop version we store on destructor only which is almost never called on Android as usually apps get swiped out or killed by OS, @samartnik could you please revert the store on destructor but we need to keep that for desktop as well.

Reply:
I restored it in destructor. @NejcZdovc  @SergeyZhukovsky is it enough to resolve this?